utils/src/treev2/quickPickWizard/quickPickAzureResource/QuickPickAppResourceStep.ts

```diff
@@ -0,0 +1,62 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { TreeItem } from 'vscode';
+import { AzureResourceQuickPickWizardContext } from '../../../../hostapi.v2';
+import * as types from '../../../../index';
+import { parseContextValue } from '../../../utils/contextUtils';
+import { PickFilter } from '../common/PickFilter';
+import { GenericQuickPickOptions, GenericQuickPickStep } from '../GenericQuickPickStep';
+import { AppResourceItem } from './tempTypes';
+
+interface AppResourceQuickPickOptions extends GenericQuickPickOptions {
+    resourceTypes?: types.AzExtResourceType[];
+    childItemFilter?: types.ContextValueFilter;
+}
+
+export class QuickPickAppResourceStep extends GenericQuickPickStep<AzureResourceQuickPickWizardContext, AppResourceQuickPickOptions> {
+    protected override async promptInternal(wizardContext: AzureResourceQuickPickWizardContext): Promise<AppResourceItem> {
+        const pickedAppResource = (await super.promptInternal(wizardContext)) as unknown as AppResourceItem;
+
+        // TODO
+        wizardContext.resource = pickedAppResource.resource;
+        wizardContext.resourceGroup = pickedAppResource.resource.resourceGroup;
+
+        return pickedAppResource;
+    }
+
+    protected readonly pickFilter: PickFilter = new AppResourcePickFilter(this.pickOptions);
+}
+
+class AppResourcePickFilter implements PickFilter {
+
+    constructor(private readonly pickOptions: AppResourceQuickPickOptions) { }
+
+    isFinalPick(node: TreeItem): boolean {
+        // If childItemFilter is defined, this cannot be a direct pick
+        if (this.pickOptions.childItemFilter) {
+            return false;
+        }
+
+        return this.matchesResourceType(parseContextValue(node.contextValue));
+    }
+
+    isAncestorPick(node: TreeItem): boolean {
+        // If childItemFilter is undefined, this cannot be an indirect pick
+        if (!this.pickOptions.childItemFilter) {
+            return false;
+        }
+
+        return this.matchesResourceType(parseContextValue(node.contextValue));
+    }
+
+    private matchesResourceType(contextValues: string[]): boolean {
+        if (!contextValues.includes('azureResource')) {

```

Comment:
Just checking: we've made sure that `BranchDataProviderItem` instances that represent application resources generates tree items with the `'azureResource'` context value (in addition to whatever the BDP generates)?

Reply:
Yes. That's definitely happening in the compatibility branch. I'm not sure if it's currently happening in the api-v2 branch yet.

Reply:
The `'azureResource'` context is put into the `options.contextValues` in `GroupingItem.getChildren`.